src/Stratis.Bitcoin.Features.Miner.Tests/PosMintingTest.cs

```diff
@@ -351,6 +352,96 @@ public void GetNetworkWeight_UsesLast73Blocks_CalculatedNetworkWeightUsingLatest
+        }
+
+        /// <summary>This is a test of coinstake age softfork activation on testnet.</summary>
+        /// <remarks><see cref="PosConsensusOptions.GetStakeMinConfirmations"/></remarks>
+        [Fact]
+        public void CoinstakeAge_AfterActivation_Testnet()
+        {
+            Assert.True(this.WasUtxoSelectedForStaking(Network.StratisTest, 450000, 450000 - 18));
+            
+            Assert.False(this.WasUtxoSelectedForStaking(Network.StratisTest, 450000, 450000 - 17));
+        }
+
+        /// <summary>This is a test of coinstake age softfork activation on testnet.</summary>
+        /// <remarks><see cref="PosConsensusOptions.GetStakeMinConfirmations"/></remarks>
+        [Fact]
+        public void CoinstakeAge_AtTheActivation_Testnet()
+        {
+            Assert.True(this.WasUtxoSelectedForStaking(Network.StratisTest, 436000 - 2, 436000 - 10)); // mining block before activation
+            
+            Assert.True(this.WasUtxoSelectedForStaking(Network.StratisTest, 436000 - 1, 436000 - 19)); // mining activation block
+            
+            Assert.False(this.WasUtxoSelectedForStaking(Network.StratisTest, 436000 - 1, 436000 - 18)); // mining activation block
+        }
+
+        /// <summary>This is a test of coinstake age softfork activation on mainnet.</summary>
+        /// <remarks><see cref="PosConsensusOptions.GetStakeMinConfirmations"/></remarks>
+        [Fact]
+        public void CoinstakeAge_BeforeActivation_Mainnet()
+        {
+            Assert.True(this.WasUtxoSelectedForStaking(Network.StratisMain, 1000, 1000 - 48)); // utxo depth is 49, mining block at 50
+            
+            Assert.False(this.WasUtxoSelectedForStaking(Network.StratisMain, 1000, 1000 - 47)); // utxo depth is 48, mining block at 49
+        }
+
+        /// <summary>This is a test of coinstake age softfork activation on mainnet.</summary>
+        /// <remarks><see cref="PosConsensusOptions.GetStakeMinConfirmations"/></remarks>
+        [Fact]
+        public void CoinstakeAge_AfterActivation_Mainnet()
+        {
+            Assert.True(this.WasUtxoSelectedForStaking(Network.StratisMain, 950000, 950000 - 498));
+            
+            Assert.False(this.WasUtxoSelectedForStaking(Network.StratisMain, 950000, 950000 - 497));
+        }
+
+        /// <summary>This is a test of coinstake age softfork activation on mainnet.</summary>
+        /// <remarks><see cref="PosConsensusOptions.GetStakeMinConfirmations"/></remarks>
+        [Fact]
+        public void CoinstakeAge_AtTheActivation_Mainnet()
+        {
+            Assert.True(this.WasUtxoSelectedForStaking(Network.StratisMain, 940000 - 2, 940000 - 50)); // mining block before activation
+            
+            Assert.True(this.WasUtxoSelectedForStaking(Network.StratisMain, 940000 - 1, 940000 - 499)); // mining activation block
```

Comment:
All the big numbers here should be constants

Reply:
Fixed